### ISSUE TITLE
[Bug] 2.16.37alpha: collection doesn't use CSS styles and pictures any more

### ISSUE THREAD

**Issue by david-allison** [ankidroid/Anki-Android#10175]
@david-allison-1  I updated to 2.16.37alpha and didn't recognize any crashes so far. But my collection now doesn't use CSS styles and pictures any more. It's like Ankidroid just ignores the CSS file. Any idea what causes that bug in 37alpha?

_Originally posted by @Anthropos888 in https://github.com/ankidroid/Anki-Android/issues/10104#issuecomment-1011479168_

**Comment by david-allison-1** [comment]
@Anthropos888 

Could you provide a sample card + media where this fails? 

Images appear to be working on my Android 11.

**Comment by mikehardy** [comment]
That could help yes (loading older versions) as we could see where the break happened roughly. I have been generating alphas still semi-frequently so that won't leave too many commits. I think you'll have to uninstall the existing parallel (or use a parallel B or something) because downgrades aren't allowed in Android package manager

**Comment by david-allison-1** [comment]
Could you try a parallel release to try and locate the broken revision without uninstalling?

I've tried `<link rel="stylesheet" href="tmp.css"/>` and it worked.

**Comment by Anthropos888** [comment]
@david-allison-1 @mikehardy 
Thanks, 35alpha is working.

Two things:
1) I installed 35alpha parallel.A first but I couldn't open my collection because it was blocked by the regular app. Closing the regular app didn't help. So whar could I do in order to open the collection with a parallel build?

2) the parallel A build also got installed as "AnkiDroid", not Ankidroid.A
There was just no way to find out which one is which one. Even the app info screen didn't tell. So I ended up uninstalling both apps.
That's confusing, I'd expect a parallel build A to have the letter .A in the app name (Android 11)

**Comment by mikehardy** [comment]
I peeled the parallel builds identification thing from item 2 above to a new issue, I noticed it myself

**Comment by david-allison-1** [comment]
> 1. I installed 35alpha parallel.A first but I couldn't open my collection because it was blocked by the regular app. Closing the regular app didn't help. So whar could I do in order to open the collection with a parallel build?

"Force closing" the app via the app picker has always worked for me. Anki's backend takes a lock of the database, and killing the process removes this lock

**Comment by david-allison-1** [comment]
@mikehardy Would a revert of https://github.com/ankidroid/Anki-Android/commit/ba416198ca79909e59ebceb0db9763f46cd249f7 be acceptable? I don't have a reproduction case, and this feels like the most likely cause

**Comment by mikehardy** [comment]
That commit was my first thought, so sure. Can always re revert if the next alpha isn't functional

**Comment by mikehardy** [comment]
@Anthropos888 took a little bit to get the release out owing to some busted translations but I got it all patched up and here's a new one for you to test, hopefully it works! :crossed_fingers: https://github.com/ankidroid/Anki-Android/releases/tag/v2.16alpha38

**Comment by Anthropos888** [comment]
@mikehardy Thanks Mike, but 37alpha and 38alpha both have the same bug. Styles from extern CSS files can't be read.
I'll uninstall 38alpha and reinstall 35alpha which works fine.

**Comment by Anthropos888** [comment]
@mikehardy Fixed with 40alpha. Thanks for your efforts!

**Comment by david-allison-1** [comment]
@Anthropos888 Fantastic!!
@mikehardy Was the fix known? I don't recall anything going in which would have caused/fixed this

**Comment by mikehardy** [comment]
Nope, this is puzzling

**Comment by Anthropos888** [comment]
@david-allison-1 @mikehardy 
I only tried 35alpha, 37alpha, 38alpha and 40alpha
1) 35alpha doesn't have the css bug, but the widget crash bug (which was fixed in 37alpha)
2) 37alpha and 38alpha both have the css bug
3) 40alpha doesn't have the css bug nor the widget bug

**Comment by david-allison-1** [comment]
Thanks!

https://github.com/ankidroid/Anki-Android/commit/f75877819fa161718b16951b4b5b023961cce060 was what I expected to fix this, and was in `38`. I'm confused

Changes from 38 to 40: https://github.com/ankidroid/Anki-Android/compare/v2.16alpha38...v2.16alpha40

**Comment by nihil-admirari** [comment]
> Yes, pictures from a card field are working, but everything that is loaded via css file fails (fonts, pictures, svg)

@Anthropos888, the first question is: how the exactly the card was exported and imported? If you have a

```html
<link rel="stylesheet" href="_styles.css">
```

in your template, and `_styles.css` references other files, e.g.

```css
@font-face {
    font-family: "Font";
    src: url("_font.woff2") format("woff2");
}

#elem {
    background-image: url("_back.png");
}
```

then when you do an export, only `_styles.css` ends up in the `.apkg`, Anki doesn't recurse into linked CSS in search for \_underscored resources. Consequently, when that `.apkg` gets imported, whatever was referenced from CSS fails to load because it's simply not present in the media folder.

However, if media folder is synced, resources will appear and everything will start working again, which leads us to the second question.

>  35alpha doesn't have the css bug, but the widget crash bug (which was fixed in 37alpha)
    37alpha and 38alpha both have the css bug
    40alpha doesn't have the css bug nor the widget bug

Were all these tests done on exactly the same collection, or the collection was synced between 38alpha and 40alpha?
